File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.java
import java.util.*;

class Solution {
    boolean solution(String s) {
        ArrayDeque<Character> stack = new ArrayDeque<>();
        for(int i = 0; i < s.length(); i++) {
            char c = s.charAt(i);
            if(c == '(') stack.add(c);
            else if(stack.isEmpty()) return false;
            else stack.removeLast();
        }
        
        return stack.isEmpty();
    }
}

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.py
def solution(s):
    st = []
    s = list(s)
    
    for i in s:
        if i == '(': 
            st.append(i)
        if i == ')':
            if len(st) == 0: return False
            st.pop()
    
    if len(st) != 0: return False
    return True